### PR TITLE
feat(tools/scripts): test project create tool

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

This moves the test block name verification system in the utils file with a slight modification by trimming the block name before running the conditional statements. With this function moved, I was in a great position to add a unit test. Thusly I added some to cover our existing cases.

### CODE REVIEW

File: tools/challenge-helper-scripts/create-project.ts [hunk 1]
```diff
@@ -198,15 +198,7 @@ void prompt([
...
-        return 'please enter a dashed name';
-      }
-      if (/[^a-z0-9-]/.test(block)) {
-        return 'please use alphanumerical characters and kebab case';
-      }
-      return true;
-    },
+    validate: validateBlockName,
     filter: (block: string) => {
       return block.toLowerCase();
```
Comment (ojeytonwilliams) [inline tools/challenge-helper-scripts/create-project.ts:203]:
```suggestion
      return block.toLowerCase().trim();
```
If we're allowing trailing whitespace to pass validation, I think we should remove it here.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
More unit tests are always welcome, thanks @a2937!

Just one nitpick:

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
I'm gonna go the opposite of Oliver here and say we should not allow trailing whitespace to pass - unless we have tests in place elsewhere that confirm the whitespace won't appear in or impact the client in any way.

**Comment by ojeytonwilliams** [comment]
Sorry, yeah, that's what I meant. We definitely shouldn't allow spaces into `block`, but if we can validate, say, `stuff-and-things `, but immediately trim it to `stuff-and-things`, that seems fine to me.

**Review by naomi-lgbt — APPROVED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Pull request merged**